jerry-main/main-unix.c

```diff
@@ -610,25 +658,47 @@ main (int argc,
       {
         static uint8_t snapshot_save_buffer[ JERRY_BUFFER_SIZE ];
 
-        size_t snapshot_size = jerry_parse_and_save_snapshot ((jerry_char_t *) source_p,
-                                                              source_size,
-                                                              is_save_snapshot_mode_for_global_or_eval,
-                                                              false,
-                                                              snapshot_save_buffer,
-                                                              JERRY_BUFFER_SIZE);
-        if (snapshot_size == 0)
+        if (is_save_snapshot_mode)
         {
-          ret_value = jerry_create_error (JERRY_ERROR_COMMON, (jerry_char_t *) "");
+          size_t snapshot_size = jerry_parse_and_save_snapshot ((jerry_char_t *) source_p,
+                                                                source_size,
+                                                                is_save_snapshot_mode_for_global_or_eval,
+                                                                false,
+                                                                snapshot_save_buffer,
+                                                                JERRY_BUFFER_SIZE);
+          if (snapshot_size == 0)
+          {
+            ret_value = jerry_create_error (JERRY_ERROR_COMMON, (jerry_char_t *) "");
+          }
+          else
+          {
+            FILE *snapshot_file_p = fopen (save_snapshot_file_name_p, "w");
+            fwrite (snapshot_save_buffer, sizeof (uint8_t), snapshot_size, snapshot_file_p);
+            fclose (snapshot_file_p);
+          }
         }
-        else
+        if (!jerry_value_has_error_flag (ret_value) && is_save_literals_mode)
         {
-          FILE *snapshot_file_p = fopen (save_snapshot_file_name_p, "w");
-          fwrite (snapshot_save_buffer, sizeof (uint8_t), snapshot_size, snapshot_file_p);
-          fclose (snapshot_file_p);
+          const size_t literal_buffer_size = jerry_parse_and_save_literals ((jerry_char_t *) source_p,
+                                                                            source_size,
+                                                                            false,
+                                                                            snapshot_save_buffer,
+                                                                            JERRY_BUFFER_SIZE,
+                                                                            is_save_literals_mode_in_c_format_or_list);
+          if (literal_buffer_size == 0)
+          {
+            ret_value = jerry_create_error (JERRY_ERROR_COMMON, (jerry_char_t *) "");
+          }
+          else
+          {
+            FILE *literal_file_p = fopen (save_literals_file_name_p, "w");
+            fwrite (snapshot_save_buffer, sizeof (uint8_t), literal_buffer_size, literal_file_p);
+            fclose (literal_file_p);
```

Comment:
I'd add this file saving code the example source code of the `jerry_parse_and_save_literal` function in API docs.